### ISSUE TITLE
OpenBSD (and possibly others) has no libcap

### ISSUE THREAD

**Issue by bket** [nzbget-ng/nzbget#31]
492bf67fa38318ac41b4e8a396041b9a4582eca3 breaks building on OpenBSD as it has no libcap, nor has it a port for it. I'm not sure about the other BSD's, but as far as I can tell FreeBSD and OSX will experience the same issue.

It seems as if Libcap{,-ng} is Linux specific.

**Comment by paul-chambers** [comment]
The API I used is in POSIX, but it seems *BSD doesn't support that part of POSIX.

From a little research, looks like they have an equivalent called 'priv' and 'ucred'.

Unfortunately I have zero programming experience with any of the BSD distributions. We'll need to find someone who does to help (and test) a BSD-specific implementation.

**Comment by feld** [comment]
Yes this is very Linux specific... I am patching this out to run on FreeBSD, but I'll see if I can find a solution.

**Comment by paul-chambers** [comment]
I plan to remove this functionality, as nzbget _doesn't need_ to run with elevated privileges and thus _shouldn't_ be run with elevated privileges.

Eventually I'll replace it with a check to see if it's running as root, and complain/exit if it is.  I'll also remove the functionality to 'switch user'.

If you're going to make changes, that would be what I'd recommend, rather than doing something BSD-specific.

Hate to repeat myself, but I have no experience working with BSD APIs, pull requests are gratefully received. Open source is supposed to be a community effort.
